SoftwareGuide/Latex/DevelopmentGuidelines/CreateAModule.tex

```diff
@@ -1085,19 +1091,80 @@ \subsubsection{C++ Runtime Tracing}
 ... continue in Python session until breakpoint is hit ...
 \end{minted}
 
+\item On a macOS operating system the LLDB debugger can be used for attaching to a
+Python process in much the same way as GDB on Linux, with a few extra security
+requirements and slightly different command syntax. The following example
+attaches to a running process and sets a breakpoint inside an \code{itk.Image}
+Python object.
+
+\begin{minted}[baselinestretch=1,fontsize=\footnotesize,linenos=false,bgcolor=ltgray]{bash}
+(venv-itk) > lldb
+(lldb) process attach -- pid 99999
+Process 99999 stopped
+* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
+    frame #0: 0x00007fff203ec656 libsystem_kernel.dylib`__select + 10
+libsystem_kernel.dylib`__select:
+->  0x7fff203ec656 <+10>: jae    0x7fff203ec660            ; <+20>
+    0x7fff203ec658 <+12>: movq   %rax, %rdi
+    0x7fff203ec65b <+15>: jmp    0x7fff203e56bd            ; cerror
+    0x7fff203ec660 <+20>: retq
+Target 0: (Python) stopped.
+
+Executable module set to "/Library/Frameworks/Python.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/Python".
+Architecture set to: x86_64h-apple-macosx-.
+
+(lldb) breakpoint set -f /path/to/ITK/Modules/Core/Common/include/itkImage.hxx --line <ln>
+Breakpoint 1: 172 locations.
+(lldb) continue
+
+... continue in Python session until breakpoint is hit ...
+\end{minted}
+
+LLDB command syntax is documented at \url{https://lldb.llvm.org/index.html}.
+
+Developers may find it necessary to examine the follow security concepts
+and requirements in order to permit \code{lldb} to attach to Python:
+
+\begin{itemize}
+
+\item The Python process must be authorized for debugging. MacOS relies on the
+process of "hardened" runtimes to mitigate security concerns, which reduces
+the ability of debuggers such as \code{lldb} and other processes to attach
+to and intercept the functions of other programs. Python distributions
+are intentionally "hardened" in this way, but additional settings can be
+enabled to allow just-in-time debugging.
+
+\item It may be necessary to enter developer mode to allow Python debugging.
+This is accomplished with the following command in the developer console:
+
+\begin{minted}[baselinestretch=1,fontsize=\footnotesize,linenos=false,bgcolor=ltgray]{bash}
+DevToolsSecurity -enable
+\end{minted}
+
+\item It may be necessary to add entitlements to the Python executable so that
+\code{lldb} can attach to the hardened process. This may be accomplished by
+updating a \code{.plist} entitlements file and setting the executable entitlements.
+
+\begin{minted}[baselinestretch=1,fontsize=\footnotesize,linenos=false,bgcolor=ltgray]{bash}
+codesign -d --entitlements :- "/path/to/python" >> "/tmp/path/to/python_entitlements.plist"
+/usr/libexec/PlistBuddy -c "Add :com.apple.security.get-task-allow bool true" "/tmp/path/to/python_entitlements.plist"
```

Comment:
Are long lines wrapped in the generated PDF, or do they go off the page?

Reply:
Good thought, looking back at the [artifacts from the previous run](https://github.com/InsightSoftwareConsortium/ITKSoftwareGuide/actions/runs/1579072802) there are several instances of lines running off the page. Will fix.